### ISSUE TITLE
Superuser installation support

### ISSUE THREAD

**Issue by apprehensions** [ImranR98/Obtainium#1791]
**Describe the feature**
Automatically install android applications as superuser (eg. `su`: KernelSU, APatch, Magisk, etc)

**Describe alternatives you've considered (if applicable)**
Suffering by pressing 'install' on every application

**Comment by apprehensions** [comment]
Shizuku and superuser `su`, Magisk, KernelSU are different.

**Comment by LightWayUp** [comment]
> Shizuku and superuser `su`, Magisk, KernelSU are different.

Indeed, but if you have Magisk, or you have KernelSU with ZygiskNext as standalone Zygisk implementation, you can use [Sui](https://github.com/RikkaApps/Sui) to provide Shizuku API that Obtainium uses.

Root installation method was removed due to unresolved issues: [Root install not working](https://github.com/ImranR98/Obtainium/issues/1200#issuecomment-1923023312), [root install fail](https://github.com/ImranR98/Obtainium/issues/1396), [root installer error](https://github.com/ImranR98/Obtainium/issues/1409). It would be nice to be able to install with plain `su`, but we will need someone who understands Android pretty well to contribute the feature.

**Comment by Mafiadoener36** [comment]
Example implementation including setting custom install source: https://github.com/fcaronte/KingInstaller

**Comment by RiRiRawrZ615** [comment]
@ImranR98 can you add support for root install back? i use kernelsu next and it would be appreciated.
